influunt-api/modules/influunt-core/app/engine/GerenciadorDeEstagios.java

```diff
@@ -365,9 +366,16 @@ private boolean naoPodeExecutarOAgendamento() {
     }
 
     private boolean podeAgendar(AgendamentoTrocaPlano agendamento) {
-        return ((this.plano.isManual() && agendamento.isSaidaDoModoManual()) ||
+        boolean result = ((this.plano.isManual() && agendamento.isSaidaDoModoManual()) ||
```

Comment:
Se a variável `agendamento` for null, vai ocorrer um NPE aqui na hora de executar o `isSaidaModoManual()`. O ideal seria verificar se `agendamento != null` antes de chamar esses métodos. E poderia escolher um nome melhor pra variável ao invés de `result`. Que tal `podeAgendar`?
```
boolean podeAgendar = angendamento != null && (this.plano.isManual() && ...)
```